### PR TITLE
Iss 236

### LINKED ISSUE #236: [Perception] Lidar point counting recognition
Publish `true` to `/perception/point_count_detection` if there are enough points in box in front of car

Tunable parameter (from launch file)
- Size of box in front of car
- Threshold number of points for detection

### LINKED ISSUE #237: [Perception] Virtual Lidar Box Visualization
Addition of a visualizer for the lidar virtual box.

Most likely going to subscribe to the `/odometry/filtered` topic with the `nav_msgs/Odometry` message and publish a `geometry_msgs/PolygonStamped` message to be published on rviz or I'll use a `visualization_msgs/Marker` message and mess with the RGBA values.

### PR DESCRIPTION
**Pull request by joeyzhu00**
Closes #236 
Closes #237

### CODE REVIEW

File: src/modules/localization/src/virtual_box_marker/virtual_box_marker.py [hunk 1]
```diff
@@ -73,15 +64,10 @@ def create_virtual_box_base_link(self):
...
-        # polygonFrontMsg.header.frame_id = "base_link"
-        # polygonRearMsg.header.frame_id = "base_link"
-        # polygonRightMsg.header.frame_id = "base_link"
-        # polygonLeftMsg.header.frame_id = "base_link"
-
-        polygonFrontMsg.header.frame_id = "velodyne"
-        polygonRearMsg.header.frame_id = "velodyne"
-        polygonRightMsg.header.frame_id = "velodyne"
-        polygonLeftMsg.header.frame_id = "velodyne"
+        polygonFrontMsg.header.frame_id = "base_link"
```
Comment (drofp) [inline src/modules/localization/src/virtual_box_marker/virtual_box_marker.py:67]: Can the Velodyne point cloud be visualized in the frame?

Comment (joeyzhu00) [reply src/modules/localization/src/virtual_box_marker/virtual_box_marker.py:67]: I'd have to create a new issue to define the `tf` from `velodyne` to `base_link`

Comment (joeyzhu00) [reply src/modules/localization/src/virtual_box_marker/virtual_box_marker.py:67]: It was being addressed in the `google_cartographer` issue, but that fell apart so I'll just make a new issue for it

Comment (drofp) [reply src/modules/localization/src/virtual_box_marker/virtual_box_marker.py:67]: sounds like a plan. will that new issue have to be merged before this one then? if so, maybe note that somewhere

Comment (joeyzhu00) [reply src/modules/localization/src/virtual_box_marker/virtual_box_marker.py:67]: Nah this pull request can proceed, I can finish up the new issue (#243) some time tomorrow

**Review by drofp — APPROVED** [review]
Validated visually with @joeyzhu00 this last weekend. Builds fine.

**Pull request merged**